spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/test/java/org/springframework/boot/gradle/junit/GradleCompatibilityExtension.java

```diff
@@ -57,6 +57,9 @@
 
 	@Override
 	public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext context) {
+		if (JavaVersion.current() == JavaVersion.VERSION_17) {
+			return Stream.of(new DisabledTemplateInvocationContext());
```

Comment:
Unfortunately, we can't just return an empty stream here. JUnit is expecting at least one invocation context